update_db & run_update_function to Friendica\Core\Update

nupplaphil opened the pull request:
I try to reduce the functions in `boot.php` step by step.
Now we have a new `Update` class.

I think, the `update.php` in the root should stay there, shouldn't it? Because otherwise we can create a Class `Friendica\Core\Update\Functions` for them.

I don't know, if I'm on the right path from your part, so please give feedback if it's better to not touching this part of Friendica without 100% knowledge of everything :-)

nupplaphil commented on src/Core/Update.php:
```diff
@@ -0,0 +1,123 @@
+	{
+		$build = Config::get('system', 'build');
+
+		if (empty($build) || ($build > DB_UPDATE_VERSION)) {
```
The `DB_UPDATE_VERSION` stays in the `boot.php` because it has to get found easily, hasn't it? That's why I don't add it as a `const` f.e. in `DBStructure` (semantically better understandable).

MrPetovan replied:
I would move `DB_UPDATE_VERSION` to `index.php`, the goal is to get rid of `boot.php`.

nupplaphil replied:
But `index.php` isn't called everytime is it?
f.e. I think, the Worker with DB-Upgrade won't execute it, will it? Just to be sure...

nupplaphil replied:
What about setting it in another `confi/*.ini.php` like `system.ini.php` togetether with the other `define()` in `boot.php` or maybe "just" inside `config.ini.php`.

MrPetovan replied:
Inside `config.ini.php` is a great idea.

MrPetovan commented on src/Core/Update.php:
```diff
@@ -0,0 +1,123 @@
+			if (!is_null($t)) {
+				return false;
+			}
+			Config::set('database', $funcname, time());
```
Same here, I want to replace those by `database.last_successful_update_function` and `database.last_successful_update_function_time`.

nupplaphil commented:
https://friendica.philipp.info started. Page works, console dbstructure update worked (but had nothing to do!) and no cache entry for "update"

MrPetovan commented on mod/admin.php:
```diff
@@ -1576,7 +1576,7 @@ function admin_page_dbsync(App $a)
 		if ($retval === '') {
 			$o .= L10n::t("Database structure update %s was successfully applied.", DB_UPDATE_VERSION) . "<br />";
-			Config::set('database', 'dbupdate_' . DB_UPDATE_VERSION, 'success');
+			Config::set('database', 'last_successful_update', time());
```
Shouldn't we also set the DB version we successfully updated to?

nupplaphil commented:
@annando what do you say about the `Update` class? Because you're the DB guy ;-)

MrPetovan commented:
Waiting for @annando's answer.

nupplaphil commented on src/Database/DBStructure.php:
```diff
@@ -22,6 +22,10 @@
  */
 class DBStructure
 {
+	const UPDATE_NOT_CHECKED = 0; // Database check wasn't executed before
```
the result code is different between DBSTRUCTURE::UPDATE_SUCCESSFUL(is 1) and UPDATE::SUCCESSFUL(is 0). I had the fear to mix the codes and therefore create bugs.

MrPetovan replied:
Ugh.

annando commented:
Sorry, I totally ignored this PR. I will have a look at it at home.

annando commented on src/Worker/DBUpdate.php:
```diff
@@ -6,17 +6,12 @@
-
-		// We are deleting the latest dbupdate entry.
-		// This is done to avoid endless loops because the update was interupted.
-		Config::delete('database', 'dbupdate_'.DB_UPDATE_VERSION);
```
Why have you removed this functionality?

MrPetovan replied:
I believe this is because he followed my suggestion to have a single config value holding the current DB update version of the Friendica node. Since we only have a one-way forward update mechanism, having multiple config keys probably didn't make sense.

nupplaphil replied:
yep, as @MrPetovan said we now have 4 config entries:
- `last_successful_update = DB_UPDATE_VERSION`
- `last_successful_update_time = time()`
- `last_successful_update_function = $funcname`
- `last_successful_update_function_time = time()`

annando replied:
So this means that we not only have to test the whole update mechanism, but also we have to test the condition that lead to this code that had been removed.

Thing is: We have to avoid that the update mechanism can be called when the updater mechanism is already running. This is crucial, since it is called automatically at page load.

But: When something is going wrong, we have to be able to override it by hand.

We had some issue here lately that was (most likely) caused by calling the structure update by command line while it already had been started from index.php. So I suggest that we check if it is running and only if you add some "--override" (or similar) you can start it again.

annando commented:
Since the update mechanism isn't easily to test on current systems, I would prefer that you install some system on "master" and then you update it to your branch. Please add some logging (or whatever) to ensure that the function ```update_1288``` is executed. Additionally please add (just for testing purposes) a function named ```pre_update_1288``` and check if it had been executed before the structure changes.

Errors here could easily lead to data loss and frustrated users - so we should test it really carefully.

nupplaphil commented:
OK, what I will do:
- backup https://opensocial.at (master 2018.09 with some users)
- adding the `pre_update_1288`
- enable debug logging
- update it to this branch

Let's have a look what happen ;-)

annando commented:
You should not only enable debug logging but add some debug logging to the update functions ;-)

annando commented:
I know that I'm very cautious about this here - but this is surely one of the most delicate parts of the system.

nupplaphil commented:
I really understand your point of view. That's why I wanted your feedback about this PR :-) .

MrPetovan commented:
More conflicts!

nupplaphil commented:
I added what @annando suggests:
- The console `dbstructure update` now uses the same locking-mechanism like the automatically update
- The lock of `dbupdate` is now infinite so we never release it automatically (avoiding double updates)
- Introduced the `--force` flag to `dbstructure update` to manually release the lock

MrPetovan commented:
If it isn't there, I would add a logger call just before each post-update function is called to verify they're called and in order.

MrPetovan commented:
Except I would remove them once the tests are conclusive not to clog the log file!

nupplaphil commented:
1) added more logger output
https://github.com/friendica/friendica/pull/5830/commits/bf878d2ebb6156b27efa9c81a6f0d10fd632922f

2) added `pre_update_1288()`
```php
function pre_update_1288() {
	\Friendica\Core\Logger::log('Pre update 1288 successfully called', \Friendica\Core\Logger::DEBUG);
}
```
3) Added PHP_errors to `boot.php`

4) executing `php bin/console.php dbstructure update`

Console Output:
```
CREATE TABLE IF NOT EXISTS `apcontact` (
        `url` varbinary(255) NOT NULL COMMENT 'URL of the contact',
        `uuid` varchar(255) COMMENT '',
        `type` varchar(20) NOT NULL COMMENT '',
        `following` varchar(255) COMMENT '',
        `followers` varchar(255) COMMENT '',
        `inbox` varchar(255) NOT NULL COMMENT '',
        `outbox` varchar(255) COMMENT '',
        `sharedinbox` varchar(255) COMMENT '',
        `nick` varchar(255) NOT NULL DEFAULT '' COMMENT '',
        `name` varchar(255) COMMENT '',
        `about` text COMMENT '',
        `photo` varchar(255) COMMENT '',
        `addr` varchar(255) COMMENT '',
        `alias` varchar(255) COMMENT '',
        `pubkey` text COMMENT '',
        `baseurl` varchar(255) COMMENT 'baseurl of the ap contact',
        `updated` datetime NOT NULL DEFAULT '0001-01-01 00:00:00' COMMENT '',
         PRIMARY KEY(`url`),
         INDEX `addr` (`addr`(32)),
         INDEX `alias` (`alias`(190)),
         INDEX `url` (`followers`(190))
) DEFAULT COLLATE utf8mb4_general_ci COMMENT='ActivityPub compatible contacts - used in the ActivityPub implementation';
ALTER TABLE `contact` ADD `deleted` boolean NOT NULL DEFAULT '0' COMMENT 'Contact has been deleted'; 
ALTER TABLE `conversation` MODIFY `protocol` tinyint unsigned NOT NULL DEFAULT 255 COMMENT 'The protocol of the item'; 
CREATE TABLE IF NOT EXISTS `diaspora-interaction` (
        `uri-id` int unsigned NOT NULL COMMENT 'Id of the item-uri table entry that contains the item uri',
        `interaction` mediumtext COMMENT 'The Diaspora interaction',
         PRIMARY KEY(`uri-id`)
) DEFAULT COLLATE utf8mb4_general_ci COMMENT='Signed Diaspora Interaction';
ALTER TABLE `item-activity` ADD INDEX `uri-id` (`uri-id`); 
ALTER TABLE `item-content` ADD INDEX `uri-id` (`uri-id`); 
ALTER TABLE `workerqueue` DROP INDEX `priority_created`, DROP INDEX `done_executed`, ADD `next_try` datetime NOT NULL DEFAULT '0001-01-01 00:00:00' COMMENT 'Next retrial date', ADD `retrial` tinyint NOT NULL DEFAULT 0 COMMENT 'Retrial counter', ADD INDEX `priority_created_next_try` (`priority`,`created`,`next_try`), ADD INDEX `done_executed_next_try` (`done`,`executed`,`next_try`); 
```

`friendica.log` Output (Level 3 - Debugging) during install :
```
2018-10-31T14:16:56Z@84ovkmve3rk1ffdkiq6487snm3	[Debug]:App.php:1276:isDirectoryUsable	Path "/tmp/opensocial.at" is not writable for user www-data
2018-10-31T14:16:56Z@84ovkmve3rk1ffdkiq6487snm3	[Debug]:App.php:1276:isDirectoryUsable	Path "/tmp/opensocial.at" is not writable for user www-data
2018-10-31T14:16:56Z@84ovkmve3rk1ffdkiq6487snm3	[Info]:DBA.php:557:p	DB Error 1054: Unknown column 'next_try' in 'where clause'
DBA::count, Worker::totalEntries, Worker::tooMuchWorkers, Worker::add, Update::check, App::runFrontend
SELECT COUNT(*) AS `count` FROM `workerqueue` WHERE (`executed` <= '0001-01-01 00:00:00' AND NOT `done` AND `next_try` < '2018-10-31 14:16:56')
2018-10-31T14:16:56Z@84ovkmve3rk1ffdkiq6487snm3	[Info]:DBA.php:557:p	DB Error 1054: Unknown column 'next_try' in 'where clause'
DBA::count, Worker::deferredEntries, Worker::tooMuchWorkers, Worker::add, Update::check, App::runFrontend
SELECT COUNT(*) AS `count` FROM `workerqueue` WHERE (`executed` <= '0001-01-01 00:00:00' AND NOT `done` AND `next_try` > '2018-10-31 14:16:56')
2018-10-31T14:16:56Z@84ovkmve3rk1ffdkiq6487snm3	[Debug]:Worker.php:706:tooMuchWorkers	Load: 0.11/50 - processes: /0/ - maximum: 20/20
2018-10-31T14:17:36Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:72:run	Update from '1283'  to '1289' - starting
2018-10-31T14:17:36Z@log:23846:5bd9b980e74c8184	[Debug]:App.php:1276:isDirectoryUsable	Path "/tmp/opensocial.at" is not writable for user vagrant
2018-10-31T14:17:36Z@log:23846:5bd9b980e74c8184	[Debug]:App.php:1276:isDirectoryUsable	Path "/tmp/opensocial.at" is not writable for user vagrant
2018-10-31T14:17:36Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:137:runUpdateFunction	Update function 'pre_update_1284' - start
2018-10-31T14:17:36Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:175:runUpdateFunction	Skipping 'pre_update_1284' without executing
2018-10-31T14:17:36Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:137:runUpdateFunction	Update function 'pre_update_1285' - start
2018-10-31T14:17:36Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:175:runUpdateFunction	Skipping 'pre_update_1285' without executing
2018-10-31T14:17:36Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:137:runUpdateFunction	Update function 'pre_update_1286' - start
2018-10-31T14:17:36Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:175:runUpdateFunction	Skipping 'pre_update_1286' without executing
2018-10-31T14:17:36Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:137:runUpdateFunction	Update function 'pre_update_1287' - start
2018-10-31T14:17:36Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:175:runUpdateFunction	Skipping 'pre_update_1287' without executing
2018-10-31T14:17:36Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:137:runUpdateFunction	Update function 'pre_update_1288' - start
2018-10-31T14:17:36Z@log:23846:5bd9b980e74c8184	[Debug]:App.php:1276:isDirectoryUsable	Path "/tmp/opensocial.at" is not writable for user vagrant
2018-10-31T14:17:36Z@log:23846:5bd9b980e74c8184	[Debug]:App.php:1276:isDirectoryUsable	Path "/tmp/opensocial.at" is not writable for user vagrant
2018-10-31T14:17:36Z@log:23846:5bd9b980e74c8184	[Debug]:update.php:261:pre_update_1288	Pre update 1288 successfully called
2018-10-31T14:17:36Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:170:runUpdateFunction	Update function 'pre_update_1288' - finished
2018-10-31T14:17:36Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:137:runUpdateFunction	Update function 'pre_update_1289' - start
2018-10-31T14:17:36Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:175:runUpdateFunction	Skipping 'pre_update_1289' without executing
2018-10-31T14:17:36Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:177:update	updating structure
2018-10-31T14:17:36Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table addon ...
2018-10-31T14:17:36Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table attach ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table auth_codes ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table cache ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table challenge ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table clients ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table config ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table contact ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table conv ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table conversation ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table event ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table fcontact ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table fsuggest ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table gcign ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table gcontact ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table glink ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table group ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table group_member ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table gserver ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table gserver-tag ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table hook ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table intro ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table item ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table item-activity ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table item-content ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table item-delivery-data ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table item-uri ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table locks ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table mail ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table mailacct ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table manage ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table notify ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table notify-threads ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table oembed ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table openwebauth-token ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table parsed_url ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table participation ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table pconfig ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table permissionset ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table photo ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table poll ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table poll_result ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table process ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table profile ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table profile_check ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table push_subscriber ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table queue ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table register ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table search ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table session ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table sign ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table term ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table thread ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table tokens ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table user ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table user-contact ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table user-item ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table userd ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table worker-ipc ...
2018-10-31T14:17:37Z@log:23846:5bd9b980e74c8184	[Debug]:DBStructure.php:190:update	updating structure for table workerqueue ...
2018-10-31T14:17:38Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:101:run	Update from '1283'  to '1289' - finished
2018-10-31T14:17:38Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:137:runUpdateFunction	Update function 'update_1284' - start
2018-10-31T14:17:38Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:175:runUpdateFunction	Skipping 'update_1284' without executing
2018-10-31T14:17:38Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:137:runUpdateFunction	Update function 'update_1285' - start
2018-10-31T14:17:38Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:175:runUpdateFunction	Skipping 'update_1285' without executing
2018-10-31T14:17:38Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:137:runUpdateFunction	Update function 'update_1286' - start
2018-10-31T14:17:38Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:175:runUpdateFunction	Skipping 'update_1286' without executing
2018-10-31T14:17:38Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:137:runUpdateFunction	Update function 'update_1287' - start
2018-10-31T14:17:38Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:175:runUpdateFunction	Skipping 'update_1287' without executing
2018-10-31T14:17:38Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:137:runUpdateFunction	Update function 'update_1288' - start
2018-10-31T14:17:38Z@log:23846:5bd9b980e74c8184	[Debug]:App.php:1276:isDirectoryUsable	Path "/tmp/opensocial.at" is not writable for user vagrant
2018-10-31T14:17:38Z@log:23846:5bd9b980e74c8184	[Debug]:App.php:1276:isDirectoryUsable	Path "/tmp/opensocial.at" is not writable for user vagrant
2018-10-31T14:17:38Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:170:runUpdateFunction	Update function 'update_1288' - finished
2018-10-31T14:17:38Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:137:runUpdateFunction	Update function 'update_1289' - start
2018-10-31T14:17:38Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:175:runUpdateFunction	Skipping 'update_1289' without executing
2018-10-31T14:17:38Z@log:23846:5bd9b980e74c8184	[Debug]:Update.php:112:run	Update from '1283'  to '1289' - successful
```

I can add the friendica.log after the first Worker call if you like.
- It has no Errors
- It pulled successfully new items

Shall I retry the install because of the missing `/tmp/opensocial.at` directory permissions?

nupplaphil commented:
Ha, I found one issue
```
2018-10-31T14:30:01Z@log:23940:5bd9bc6944f3b970	[Debug]:DBA.php:416:p	Parameter mismatch. Query "SELECT `uid`, `language`, `email` FROM `user` WHERE (`email` IN (%s))" - Parameters Array
(
    [1] => 'admin@friendica.local'
)

2018-10-31T14:30:01Z@log:23940:5bd9bc6944f3b970	[Info]:DBA.php:557:p	DB Error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '%s))' at line 1
DBA::select, Update::updateSuccessfull, Update::run, Update::check
SELECT `uid`, `language`, `email` FROM `user` WHERE (`email` IN (%s))
2018-10-31T14:30:01Z@log:23940:5bd9bc6944f3b970	[Trace]:Update.php:264:updateSuccessfull	Database structure update successful.
```

=> fixing it and retry it again

nupplaphil commented:
This time with right permissions :-)

Console output:
```
CREATE TABLE IF NOT EXISTS `apcontact` (
        `url` varbinary(255) NOT NULL COMMENT 'URL of the contact',
        `uuid` varchar(255) COMMENT '',
        `type` varchar(20) NOT NULL COMMENT '',
        `following` varchar(255) COMMENT '',
        `followers` varchar(255) COMMENT '',
        `inbox` varchar(255) NOT NULL COMMENT '',
        `outbox` varchar(255) COMMENT '',
        `sharedinbox` varchar(255) COMMENT '',
        `nick` varchar(255) NOT NULL DEFAULT '' COMMENT '',
        `name` varchar(255) COMMENT '',
        `about` text COMMENT '',
        `photo` varchar(255) COMMENT '',
        `addr` varchar(255) COMMENT '',
        `alias` varchar(255) COMMENT '',
        `pubkey` text COMMENT '',
        `baseurl` varchar(255) COMMENT 'baseurl of the ap contact',
        `updated` datetime NOT NULL DEFAULT '0001-01-01 00:00:00' COMMENT '',
         PRIMARY KEY(`url`),
         INDEX `addr` (`addr`(32)),
         INDEX `alias` (`alias`(190)),
         INDEX `url` (`followers`(190))
) DEFAULT COLLATE utf8mb4_general_ci COMMENT='ActivityPub compatible contacts - used in the ActivityPub implementation';
ALTER TABLE `contact` ADD `deleted` boolean NOT NULL DEFAULT '0' COMMENT 'Contact has been deleted'; 
ALTER TABLE `conversation` MODIFY `protocol` tinyint unsigned NOT NULL DEFAULT 255 COMMENT 'The protocol of the item'; 
CREATE TABLE IF NOT EXISTS `diaspora-interaction` (
        `uri-id` int unsigned NOT NULL COMMENT 'Id of the item-uri table entry that contains the item uri',
        `interaction` mediumtext COMMENT 'The Diaspora interaction',
         PRIMARY KEY(`uri-id`)
) DEFAULT COLLATE utf8mb4_general_ci COMMENT='Signed Diaspora Interaction';
ALTER TABLE `item-activity` ADD INDEX `uri-id` (`uri-id`); 
ALTER TABLE `item-content` ADD INDEX `uri-id` (`uri-id`); 
ALTER TABLE `workerqueue` DROP INDEX `priority_created`, DROP INDEX `done_executed`, ADD `next_try` datetime NOT NULL DEFAULT '0001-01-01 00:00:00' COMMENT 'Next retrial date', ADD `retrial` tinyint NOT NULL DEFAULT 0 COMMENT 'Retrial counter', ADD INDEX `priority_created_next_try` (`priority`,`created`,`next_try`), ADD INDEX `done_executed_next_try` (`done`,`executed`,`next_try`); 
```

`friendica.log`:
```
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:72:run	Update from '1283'  to '1289' - starting
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:137:runUpdateFunction	Update function 'pre_update_1284' - start
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:175:runUpdateFunction	Skipping 'pre_update_1284' without executing
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:137:runUpdateFunction	Update function 'pre_update_1285' - start
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:175:runUpdateFunction	Skipping 'pre_update_1285' without executing
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:137:runUpdateFunction	Update function 'pre_update_1286' - start
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:175:runUpdateFunction	Skipping 'pre_update_1286' without executing
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:137:runUpdateFunction	Update function 'pre_update_1287' - start
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:175:runUpdateFunction	Skipping 'pre_update_1287' without executing
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:137:runUpdateFunction	Update function 'pre_update_1288' - start
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:update.php:261:pre_update_1288	Pre update 1288 successfully called
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:170:runUpdateFunction	Update function 'pre_update_1288' - finished
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:137:runUpdateFunction	Update function 'pre_update_1289' - start
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:175:runUpdateFunction	Skipping 'pre_update_1289' without executing
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:177:update	updating structure
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table addon ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table attach ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table auth_codes ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table cache ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table challenge ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table clients ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table config ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table contact ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table conv ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table conversation ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table event ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table fcontact ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table fsuggest ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table gcign ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table gcontact ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table glink ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table group ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table group_member ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table gserver ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table gserver-tag ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table hook ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table intro ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table item ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table item-activity ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table item-content ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table item-delivery-data ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table item-uri ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table locks ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table mail ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table mailacct ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table manage ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table notify ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table notify-threads ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table oembed ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table openwebauth-token ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table parsed_url ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table participation ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table pconfig ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table permissionset ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table photo ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table poll ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table poll_result ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table process ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table profile ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table profile_check ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table push_subscriber ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table queue ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table register ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table search ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table session ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table sign ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table term ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table thread ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table tokens ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table user ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table user-contact ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table user-item ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table userd ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table worker-ipc ...
2018-10-31T14:54:09Z@log:24175:5bd9c21177b4c114	[Debug]:DBStructure.php:190:update	updating structure for table workerqueue ...
2018-10-31T14:54:11Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:101:run	Update from '1283'  to '1289' - finished
2018-10-31T14:54:11Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:137:runUpdateFunction	Update function 'update_1284' - start
2018-10-31T14:54:11Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:175:runUpdateFunction	Skipping 'update_1284' without executing
2018-10-31T14:54:11Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:137:runUpdateFunction	Update function 'update_1285' - start
2018-10-31T14:54:11Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:175:runUpdateFunction	Skipping 'update_1285' without executing
2018-10-31T14:54:11Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:137:runUpdateFunction	Update function 'update_1286' - start
2018-10-31T14:54:11Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:175:runUpdateFunction	Skipping 'update_1286' without executing
2018-10-31T14:54:11Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:137:runUpdateFunction	Update function 'update_1287' - start
2018-10-31T14:54:11Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:175:runUpdateFunction	Skipping 'update_1287' without executing
2018-10-31T14:54:11Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:137:runUpdateFunction	Update function 'update_1288' - start
2018-10-31T14:54:11Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:170:runUpdateFunction	Update function 'update_1288' - finished
2018-10-31T14:54:11Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:137:runUpdateFunction	Update function 'update_1289' - start
2018-10-31T14:54:11Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:175:runUpdateFunction	Skipping 'update_1289' without executing
2018-10-31T14:54:11Z@log:24175:5bd9c21177b4c114	[Debug]:Update.php:112:run	Update from '1283'  to '1289' - successful
```

build:
```
+----+--------+-------+------+
| id | cat    | k     | v    |
+----+--------+-------+------+
|  1 | system | build | 1289 |
+----+--------+-------+------+
1 row in set (0.00 sec)
```

new update values:
```
+-----+----------+--------------------------------------+-------------+
| id  | cat      | k                                    | v           |
+-----+----------+--------------------------------------+-------------+
| 196 | database | last_successful_update_function      | update_1289 |
| 197 | database | last_successful_update_function_time | 1540997651  |
| 198 | database | last_successful_update               | 1289        |
| 199 | database | last_successful_update_time          | 1540997651  |
+-----+----------+--------------------------------------+-------------+
4 rows in set (0.00 sec)
```

MrPetovan commented:
Looks good, thanks!
